source/module_cell/module_symmetry/symmetry.cpp

```diff
@@ -1405,21 +1398,23 @@ void Symmetry::rhog_symmetry(std::complex<double> *rhogtot,
                                 //std::cout<<"warning: ipw0 is in pw-sphere but ipw not !!!"<<std::endl;
                             continue;   //else, just skip it
                         }
+                        //fft-grid index to new-gdirect
+                        tmp_gdirect_double.x=static_cast<double>((ii>int(nx/2)+1)?(ii-nx):ii);
+                        tmp_gdirect_double.y=static_cast<double>((jj>int(ny/2)+1)?(jj-ny):jj);
+                        tmp_gdirect_double.z=static_cast<double>((kk>int(nz/2)+1)?(kk-nz):kk);
                         //calculate phase factor
-                        tmp_gdirect_double.x=(double)tmp_gdirect.x;
-                        tmp_gdirect_double.y=(double)tmp_gdirect.y;
-                        tmp_gdirect_double.z=(double)tmp_gdirect.z;
+                        tmp_gdirect_double = tmp_gdirect_double * ModuleBase::TWO_PI;
                         double cos_arg=0.0, sin_arg=0.0;
                         // for each pricell in supercell:
                         for(int ipt=0;ipt<this->ncell;++ipt)
                         {
-                            arg = ( tmp_gdirect_double * (gtrans[invmap[isym]]+ptrans[ipt]) ) * ModuleBase::TWO_PI;
+                            arg = tmp_gdirect_double * (gtrans[invmap[isym]]+ptrans[ipt]) ;
                             cos_arg += cos(arg);
```

Comment:
I guess the most expensive code is sin() and cos(), have you tested the improvement of this PR?

Reply:
#1780 can help for this.